### ISSUE TITLE
[Request] Force Enable Controller Ports

### ISSUE THREAD

**Issue by DerpyBubblez** [FIX94/Nintendont#764]
This is a bit of a strange thing to request, but there's been an influx of 2 Player codes recently. The games work properly, but the second player is unable to move their character. It works fine in Dolphin, and (I don't remember if this is theory or not) Dolphin enables all four controller ports regardless of whether or not the game needs them enabled. If we had the option to force the controller ports to be enabled, then we could use the multiplayer patch as normal.

tl;dr This would let us use multiplayer cheats on console. Currently this benefits Luigi's Mansion and Animal Crossing.

**Comment by lukehasawii** [comment]
Have there been any updates to this concept? I would love to see this happen!

**Comment by lukehasawii** [comment]
I attempted to force the ports manually, and I was able to track down quite a few different files that looked promising. Specifically, \loader\source\ppc\PADReadGC\source\PADReadGC.c and loader\source\Patches.c which has a patch for something along the lines of "Force use all controllers." My source code edit didn't yield any results (the game I am testing is Luigi's Mansion) as the cheat code succeeded in creating the two Luigis, but the second Luigi is still not playable. If anyone wants to take a look at what I did and see if they can make any further changes, please do! I didn't think to fork the repo so here's a zip file of what I was working with.[https://github.com/lukehasawii/gcforcecontrollernintendont/raw/master/Nintendont.zip]( https://github.com/lukehasawii/gcforcecontrollernintendont/raw/master/Nintendont.zip)
